### PR TITLE
Feature/add new oracle

### PR DESCRIPTION
**Pull request by tienshaoku**
### PR Reminders

Be ware of the followings

- [ ] implement deployment script for your changes in deployment repo
- [ ] add verification in hardhat simulation(000-prepare-simulation-check, 902-newMarket-check or 903-simulation-check) and system test(901-system-test) in deployment repo
- [ ] update change log
 
---

1. **Contract**: make sure the code follows our convention, ex: naming, explicit returns, etc.; if uncertain, discuss with others

2. **Test**: make sure tests can cover most normal and edge cases; if not, open follow-up tickets. Also, look out for failed tests and fix them!

3. **CHANGELOG.md**: update when external interfaces are changed

4. Workflow: 
    - Github: assign the pr to yourself; if pairing, can merge directly; else, assign someone to help review
    - Asana: assign the corresponding ticket to yourself and leave the pr link on it for easier follow-ups
    - Discord: if someone is mentioned in this pr, tag on Discord

### CODE REVIEW

File: contracts/ClearingHouseConfig.sol [hunk 1]
```diff
@@ -63,9 +63,6 @@ contract ClearingHouseConfig is IClearingHouseConfig, SafeOwnable, ClearingHouse
     }
 
     function setTwapInterval(uint32 twapIntervalArg) external onlyOwner {
-        // CHC_ITI: invalid twapInterval
-        require(twapIntervalArg != 0, "CHC_ITI");
```
Comment (tailingchen) [inline contracts/ClearingHouseConfig.sol:67]: If we want to set _twapInterval to zero, that means the funding payment will be `(marketTwap(0)-indexTwap(0)) / duration`. Is it expected?

Comment (tailingchen) [reply contracts/ClearingHouseConfig.sol:67]: As discussed, yes, it's expected. If we do not reduce the market twap interval together, it may happen that the founding rate is opposite to the expected situation. For example, the founding rate is positive, but current index price is less than market price. In this way, traders have no incentive to open long positions to push market price to be close to index price.

**Review by Nathaniel2030 — DISMISSED** [review]

File: test/clearingHouse/ClearingHouse.newMarket.ts [hunk 2]
```diff
@@ -190,7 +186,7 @@ describe("ClearingHouse new market listing", () => {
             expect(davisPendingFundingAfter.abs()).to.be.gt(davisPendingFundingBefore.abs())
         })
 
-        it("Stop to cumulate funding after change to emergency oracle", async () => {
+        it.skip("Stop to cumulate funding after change to emergency oracle", async () => {
```
Comment (phchan9) [inline test/clearingHouse/ClearingHouse.newMarket.ts]: Why we skip this test case here? Based on the new mechanism, we still might do update the priceFeed of baseToken with EmergencyPriceFeed?

Comment (phchan9) [reply test/clearingHouse/ClearingHouse.newMarket.ts]: @twxia any thoughts?

Comment (twxia) [reply test/clearingHouse/ClearingHouse.newMarket.ts]: This test is to ensure that after the emergencyPriceFeed get plugged in, the funding rate should not get cumulated. 

Original oracle:
Because mark price (market twap) = index price (emergencyPriceFeed's market twap)

New oracle:
After oracle get switched to uniswap market price. 
mark price (mid(30min-twap, 15min-twap, market price)) != index price (Uniswapv3 30 minutes market twap)
Reason is at https://www.notion.so/perp/TWAP-interval-from-15m-0s-cb5469cfc65a47faad1c6722ebc71dc7#d5d0ed9c98324e40bd0e1256a7884fcf

Comment (phchan9) [reply test/clearingHouse/ClearingHouse.newMarket.ts]: Understood that this test case is for the case when emergencyPriceFeed get plugged in. Then should we remove this case cuz we don't use emergencyPriceFeed anymore?
